src/account_multisig.ts

```diff
@@ -205,32 +212,33 @@ export class Account2FA extends AccountMultisig {
     // default helpers for CH deployments of multisig
 
     async deployMultisig(contractBytes: Uint8Array) {
-        const { accountId } = this;
-
-        const seedOrLedgerKey = (await this.getRecoveryMethods()).data
-            .filter(({ kind, publicKey }) => (kind === 'phrase' || kind === 'ledger') && publicKey !== null)
-            .map((rm) => rm.publicKey);
-
-        const fak2lak = (await this.getAccessKeys())
-            .filter(({ public_key, access_key: { permission } }) => permission === 'FullAccess' && !seedOrLedgerKey.includes(public_key))
-            .map((ak) => ak.public_key)
-            .map(toPK);
-
-        const confirmOnlyKey = toPK((await this.postSignedJson('/2fa/getAccessKey', { accountId })).publicKey);
-
-        const newArgs = Buffer.from(JSON.stringify({ 'num_confirmations': 2 }));
-
-        const actions = [
-            ...fak2lak.map((pk) => deleteKey(pk)),
-            ...fak2lak.map((pk) => addKey(pk, functionCallAccessKey(accountId, MULTISIG_CHANGE_METHODS, null))),
-            addKey(confirmOnlyKey, functionCallAccessKey(accountId, MULTISIG_CONFIRM_METHODS, null)),
-            deployContract(contractBytes),
-        ];
-        if ((await this.state()).code_hash === '11111111111111111111111111111111') {
-            actions.push(functionCall('new', newArgs, MULTISIG_GAS, MULTISIG_DEPOSIT),);
+        if (ALLOW_2FA_ENABLE_HASHES.includes((await this.state()).code_hash)) {
+            const { accountId } = this;
+            const seedOrLedgerKey = (await this.getRecoveryMethods()).data
+                .filter(({ kind, publicKey }) => (kind === 'phrase' || kind === 'ledger') && publicKey !== null)
+                .map((rm) => rm.publicKey);
+
+            const fak2lak = (await this.getAccessKeys())
+                .filter(({ public_key, access_key: { permission } }) => permission === 'FullAccess' && !seedOrLedgerKey.includes(public_key))
+                .map((ak) => ak.public_key)
+                .map(toPK);
+
+            const confirmOnlyKey = toPK((await this.postSignedJson('/2fa/getAccessKey', { accountId })).publicKey);
+
+            const newArgs = Buffer.from(JSON.stringify({ 'num_confirmations': 2 }));
+
+            const actions = [
+                ...fak2lak.map((pk) => deleteKey(pk)),
+                ...fak2lak.map((pk) => addKey(pk, functionCallAccessKey(accountId, MULTISIG_CHANGE_METHODS, null))),
+                addKey(confirmOnlyKey, functionCallAccessKey(accountId, MULTISIG_CONFIRM_METHODS, null)),
+                deployContract(contractBytes),
+                functionCall('new', newArgs, MULTISIG_GAS, MULTISIG_DEPOSIT)
+            ];
+            console.log('deploying multisig contract for', accountId);
+            return await super.signAndSendTransactionWithAccount(accountId, actions);
+        } else {
```

Comment:
nit: might be cleaner to do early exit when checking `if (!ALLOW_2FA_ENABLE_HASHES.includes((await this.state()).code_hash)) {`, so that no need to put everything inside extra block

Reply:
Good one:)